pkg/sys/stats_solaris.go

```diff
@@ -0,0 +1,41 @@
+/*
+ * Minio Cloud Storage, (C) 2016,2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sys
+
+import (
+	"os"
+)
+
+/*
+#include <unistd.h>
+*/
+import (
+	"C"
```

Comment:
Generally as a practice we would like to avoid this, because this causes requirements of `cgo`. Minio is generally built with CGO_ENABLED=0 and we want to keep it that way for simplicity and avoid depending on libc altogether. 

There might be some limitations on how much Solaris is supported but currently - you can see an example on how it should be implemented here 

https://github.com/golang/sys/blob/master/unix/syscall_solaris.go#L706